### ISSUE TITLE
Crash NPE at BranchPostTask

### ISSUE THREAD

**Issue by ismetdegoo** [BranchMetrics/android-branch-deep-linking-attribution#913]
Hi! We updated Branch version from `5.0.4` to `5.0.9` and we started getting the crash log below.
We are using deeplink, and it works fine. But there are around 10K logs every day.

Branch.io version: `5.0.9`

Error log:
```
Caused by java.lang.NullPointerException
Attempt to invoke virtual method 'void io.branch.referral.Branch.addExtraInstrumentationData(java.lang.String, java.lang.String)' on a null object reference
    io.branch.referral.BranchPostTask.doInBackground (BranchPostTask.java:39)
    io.branch.referral.BranchPostTask.doInBackground (BranchPostTask.java:17)
    android.os.AsyncTask$2.call (AsyncTask.java:292)
    java.util.concurrent.FutureTask.run (FutureTask.java:237)
    java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1112)
    java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:587)
    java.lang.Thread.run (Thread.java:818)
```

Thanks in advance!

**Comment by echo-branch** [comment]
Thanks for the heads up. Looking into reproducing it.

**Comment by ismetdegoo** [comment]
@echo-branch I should mention that we call `Branch.getAutoInstance(application)` in a background thread in `onCreate` method of application class.
In addition to above logs, here are some other logs from another thread:
```
java.lang.Object.wait (Object.java)
java.lang.Thread.parkFor$ (Thread.java:2137)
sun.misc.Unsafe.park (Unsafe.java:358)
java.util.concurrent.locks.LockSupport.parkNanos (LockSupport.java:230)
java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedNanos (AbstractQueuedSynchronizer.java:1061)
java.util.concurrent.locks.AbstractQueuedSynchronizer.tryAcquireSharedNanos (AbstractQueuedSynchronizer.java:1352)
java.util.concurrent.CountDownLatch.await (CountDownLatch.java:278)
io.branch.referral.Branch.awaitTimedBranchPostTask (Branch.java:1705)
io.branch.referral.Branch.executeTimedBranchPostTask (Branch.java:1699)
io.branch.referral.Branch.processNextQueueItem (Branch.java:1671)
io.branch.referral.Branch.onAdsParamsFetchFinished (Branch.java:972)
io.branch.referral.SystemObserver.setFireAdId (SystemObserver.java:444)
io.branch.referral.SystemObserver.prefetchAdsParams (SystemObserver.java:418)
io.branch.referral.Branch.<init> (Branch.java:402)
io.branch.referral.Branch.initBranchSDK (Branch.java:425)
io.branch.referral.Branch.getAutoInstance (Branch.java:455)
REDACTED_APP_CLASS.lambda$onCreate$4 (REDACTED_APP_CLASS:REDACTED_LINE)
REDACTED_APP_CLASS.lambda$onCreate$4$REDACTED_APP_CLASS (REDACTED_APP_CLASS)
REDACTED_PACKAGE.-$$Lambda$REDACTED_APP_CLASS$EZyx6YymDWAfXeyQGM9-yEySIw0.run (-.java:6)
REDACTED_PACKAGE.BackgroundTaskExecutor$execute$1.invokeSuspend (BackgroundTaskExecutor.java:REDACTED_LINE)
kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith (BaseContinuationImpl.java:33)
kotlinx.coroutines.DispatchedTask.run (DispatchedTask.java:106)
kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely (CoroutineScheduler.java:571)
kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask (CoroutineScheduler.java:750)
kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker (CoroutineScheduler.java:678)
kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run (CoroutineScheduler.java:665)
```

I hope this helps.

**Comment by matteinn** [comment]
I'm facing the same issue but can't replicate, just noticing a large amount of crashes in production. Is there anything we can do to help?

**Comment by ismetdegoo** [comment]
@echo-branch have you been able to solve this issue? We are still getting thousands of crash logs and Crash-free users count is decreasing.

**Comment by ateslovan** [comment]
I'm facing the same issue, same exception, but on my side it's only happening for Amazon devices (KFONWI, KFKAWI,  KFMAWI, KFMUWI).

In addition to this stacktrace, we also have another one: 

```
Fatal Exception: java.lang.NullPointerException: Attempt to write to field 'int io.branch.referral.Branch.c' on a null object reference
       at io.branch.referral.BranchPostTask.onRequestFailed(BranchPostTask.java:3)
       at io.branch.referral.Branch.awaitTimedBranchPostTask(Branch.java:28)
       at io.branch.referral.Branch.access$100(Branch.java)
       at io.branch.referral.Branch$1.run(Branch.java:8)
       at java.lang.Thread.run(Thread.java:764)
```

Branch version: io.branch.sdk.android:library:5.0.8

**Comment by Amagi82** [comment]
It's happening for us too. Can confirm this only affects Amazon devices (KFONWI & KFKAWI in my logs). Both appear to be running Android 9. Using v5.0.9 of the Branch library.

Last version of our app used v5.0.7 of Branch and had no such issues.

```
Branch.java line 2469
io.branch.referral.Branch.addExtraInstrumentationData


Fatal Exception: java.lang.RuntimeException: An error occurred while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:354)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
       at java.util.concurrent.FutureTask.run(FutureTask.java:271)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)
       
Caused by java.lang.NullPointerException: Attempt to read from field 'java.util.concurrent.ConcurrentHashMap l.a.a.c.q' on a null object reference
       at io.branch.referral.Branch.addExtraInstrumentationData(Branch.java:2469)
       at io.branch.referral.BranchPostTask.doInBackground(BranchPostTask.java:39)
       at io.branch.referral.BranchPostTask.doInBackground(BranchPostTask.java:17)
       at android.os.AsyncTask$2.call(AsyncTask.java:333)
       at java.util.concurrent.FutureTask.run(FutureTask.java:266)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
       at java.lang.Thread.run(Thread.java:764)
```

**Comment by ivan-at-chipotle** [comment]
Experiencing this same issue with the same stack trace. Whatever's happening has got to be related to something in the background, user initiated or not - we're seeing these events in the hundreds, which tells me something is either Looping as a retry mechanism on these devices, or the SDK has some more subtle background issue that's exposed in a similar way.

**Comment by jf-branch** [comment]
This NPE in `BranchPostTask.doInBackground() ` should be fixed by our [5.0.12 release](https://github.com/BranchMetrics/android-branch-deep-linking-attribution/releases/tag/5.0.12). Please update your SDKs to the newest version. We will be closing out this issue thread for now. Please reopen if the NPE persists.
